api/src/test/java/org/openmrs/AllergyTest.java

```diff
@@ -0,0 +1,269 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class AllergyTest {
+
+	@Test(expected = NullPointerException.class)
+	public void hasSameValues_otherAllergyIsNull_shouldThrowNullPointerException() {
+		Allergy allergy = new Allergy();
+		allergy.hasSameValues(null);
+	}
+
+	@Test
+	public void hasSameValues_checkingAgainstSameInstanceOfAllergy_shouldInterpretAsSameValues() {
+		Allergy allergy = allergy();
+		allergy.hasSameValues(allergy());
```

Comment:
Shouldn't this line be AssertTrue(allergy.hasSameValues(allergy())); ?

Reply:
ya yor actually right. i just missed it out . thanks @HelioStrike